crates/swift-bridge-ir/src/codegen/codegen_tests/string_codegen_tests.rs

```diff
@@ -215,3 +215,50 @@ struct RustStr __swift_bridge__$some_function(void);
         .test();
     }
 }
+
+/// Test code generation for Swift function that returns an owned String argument.
+mod extern_swift_func_returns_string {
+    use super::*;
+
+    fn bridge_module_tokens() -> TokenStream {
+        quote! {
+            mod foo {
+                extern "Swift" {
+                    fn some_function () -> String;
+                }
+            }
+        }
+    }
+
+    fn expected_rust_tokens() -> ExpectedRustTokens {
+        ExpectedRustTokens::Contains(quote! {
+            pub fn some_function () -> String {
+                unsafe {
+                    Box::from_raw(unsafe { __swift_bridge__some_function() }).0
```

Comment:
Is this really safe? You're assuming that Rust can free the memory safely, when the Swift runtime and the Rust runtime can technically have different allocators? http://jakegoulding.com/rust-ffi-omnibus/string_return/

Reply:
Yes it should be safe because:

1. `__swift_bridge__$some_function` creates a `RustString`, which is a Swift class that holds a pointer to a new type wrapper around a `std::string::String` https://github.com/chinedufn/swift-bridge/blob/23d60feaa0f8e92f2c854d8665868aa5573ef505/src/std_bridge/string.rs#L22-L23

2. Pointer to the Rust `std_bridge::string::RustString` gets passed from Swift -> Rust.

3. Rust uses `Box::from_raw` to get the `RustString(std::string::String)`

4. Rust now has ownership of the `String`.

So, the `RustString` is only ever allocated and freed from Rust's allocator.

---

Let me know if this doesn't answer your question.